lesson14/main.js

```diff
@@ -0,0 +1,122 @@
+const openButton = document.getElementById("js-open-modal-button");
+const modal = document.getElementById("js-modal");
+const closeButton = document.getElementById("js-close-button");
+const overlay = document.getElementById("js-overlay");
+const openListButton = document.getElementById("js-open-list-button")
+const ul = document.getElementById("js-ul");
+const input =document.getElementById("js-number-input")
+
+const renderLoading = () => {
+  const loading = document.getElementById("js-loading");
+  const loadingImage = document.createElement("img");
+  loadingImage.src = "img/loading.gif";
+  loadingImage.id = "loading-image";
+  loading.appendChild(loadingImage);
+};
+
+const removeLoading = () => document.getElementById("loading-image").remove();
+
+const renderLists = (items) => {
+  const fragment = new DocumentFragment();
+
+  for (let item of items) {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+    a.href = item.to;
+    a.textContent = item.text;
+    img.src = item.img;
+    img.alt = item.alt;
+
+    fragment
+      .appendChild(li)
+      .appendChild(a)
+      .insertAdjacentElement("afterbegin", img);
+  }
+
+  ul.appendChild(fragment);
+};
+
+const displayInfo = (info) => {
+  const li = document.createElement('li');
+  if (typeof info === 'string') {
+    li.textContent = info;
+    ul.appendChild(li);
+    return console.log(info);
+  }
+  throw new Error('Please pass a string value');
+};
+
+const displayErrorInfo = (error) => {
+  const li = document.createElement('li');
+  li.textContent = `error: ${JSON.stringify(error)}`;
+  ul.appendChild(li);
+};
+
+const getData = async (api) => {
+  let res = null;
+  try {
+    res = await fetch(api);
+  } catch (e) {
+    displayErrorInfo(e);
+  }
+  if (res.ok) {
+    return res.json();
+  } else {
+    displayErrorInfo(res);
+  }
+};
+
+const displayList = async () => {
+  renderLoading();
+  let res = await getData("https://mocki.io/v1/ee8a871e-2b46-4a91-b565-4d6f9216f300");
+  removeLoading();
+
+  if (res.data) {
+    renderLists(res.data);
+  } else {
+    displayInfo("no data");
```

Comment:
Q
Sorry for my lack of understanding, but one question:bow:
I did not understand the role of displayInfo().
I thought that if you specify a string as an argument, it does not have to be a function.
```suggestion
    document.createElement('li').textContent = "no data";
    console.log("no data");
```

Reply:
Thank you for your comment Yoshikawa-san.
The purpose of displayInfo() function is to handle an unexpected errors and display the error in console and UI. 
getData() function can check if the response returns ok or not. However, it does not if the response has "data" value in it, so that is where displayInfo() comes in. 

Although it is only used when response does not contain "data" in this lesson, but I believe it will be more useful as the lessons go on, because it can handle error messages in different scenarios. 

Hope it makes sense.

Reply:
Thanks for the answer!
I understand now!